ocp_resources/user_defined_network.py

```diff
@@ -66,3 +67,65 @@ def to_dict(self) -> None:
                 _spec["localNet"] = self.local_net
 
     # End of generated code
+
+
+LAYER2: str = "Layer2"
+
+
+class Layer2UserDefinedNetwork(UserDefinedNetwork):
+    """
+    UserDefinedNetwork layer2 object.
+
+    API reference:
+    https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#layer2config
+    """
+
+    def __init__(
+        self,
+        role: str,
```

Comment:
As I commented before, all args must be Optional since you can pass yaml_file or kind_dict which include all data including the `role` arg and so on.

Reply:
So I didn't understand your comment, if I will set it with default value it will be treated as optional, and then no exception raise for missing value....
Since I delete the if statement that checked it and raised the exception.

Reply:
Either way I refactored it to set it as optional.